packages/botonic-nlp/src/codify/codifier.ts

```diff
@@ -0,0 +1,39 @@
+import { unique } from '../utils/array-utils'
+
+export class Codifier {
```

Comment:
split into Encoder & Decoder classes. Rationale:
* I've been skimming other liibraries and it looks like they always go independently.
* encoder & decoder share very little state
* it will allow extending each Encoder & Decoder independently

You could create a shared class Vocabulary (but don't inherit from it, compose it) where you call unique and held the isValidToken and isValidTokenId methods

Reply:
I suggest using the vocabulary from https://www.analyticsvidhya.com/blog/2020/03/one-hot-encoding-vs-label-encoding-using-scikit-learn/ https://towardsdatascience.com/categorical-encoding-using-label-encoding-and-one-hot-encoder-911ef77fb5bd (label encoder vs one-hot encoder), which is what sklearn uses

Reply:
LGTM, I will implement the Vocabulary class in another PR. 👍🏽